### PR TITLE
dqlite support

### PR DESCRIPTION
**Pull request by ibuildthecloud**
This enables full embedded HA support by using [dqlite](https://github.com/canonical/dqlite)

```sh
# Start one server with clustering enabled on serve1
./k3s server --cluster-init --token SECRET

# On server2 join the cluster
./k3s server -t SECRET -s https://server1:6443

# On server3 join the cluster
./k3s server -t SECRET -s https://server1:6443
```

And you now have full HA.

Before you get too excited this is going to take a bit of time to bake to iron out the bugs and what not.  We will merge this as an experimental feature for a while so people can test it out.

### CODE REVIEW

**Comment by sandys** [comment]
can the servers themselves be used to run workloads ? just a cost saving measure. 

Docker Swarm allows services to be scheduled on masters as well. Makes for cost-saving.

**Comment by ibuildthecloud** [comment]
Yeah, the usecase here is to be able to run small HA 3 node clusters on the edge. So all nodes are masters and workers.

**Comment by alexellis** [comment]
It'd be great to get a PR to k3sup from Rancher to help people out with this use-case. Let me know when an RC build is available so we can test out the flow.

https://github.com/alexellis/k3sup/issues/93

The flow is currently:

```
k3sup install --ip $IP --user $USER
k3sup join --ip $IP --user $USER --server-ip $SERVER_IP
```

We could adapt this to take in the new variables

**Comment by andyjeffries** [comment]
Likewise, when it’s in RC we can add it as a development choice in Civo.com’s managed k3s service.

File: Dockerfile.dapper [hunk 1]
```diff
@@ -1,9 +1,7 @@
 FROM golang:1.12.12-alpine3.10
 
 RUN apk -U --no-cache add bash git gcc musl-dev docker vim less file curl wget ca-certificates jq linux-headers zlib-dev tar zip squashfs-tools npm coreutils \
-    python3 py3-pip python3-dev openssl-dev libffi-dev libseccomp libseccomp-dev make
-RUN pip3 install 'tox==3.6.0'
-RUN apk -U --no-cache --repository http://dl-3.alpinelinux.org/alpine/edge/main/ add sqlite-dev sqlite-static
```
Comment (yamt) [inline Dockerfile.dapper]: is python3 removal here related to dqlite?

Comment (ibuildthecloud) [reply Dockerfile.dapper]: Nope, just saw that we didn't need them

**Comment by alexellis** [comment]
Please see issue #1155 which I encountered on first attempt. Is this new mode/flag still able to use `--tls-san`?

**Comment by davidnuzik** [comment]
Please use https://github.com/rancher/k3s/issues/1115 to track instead. Thank you!

**Pull request merged**